api/v1alpha1/ext_authz_type.go

```diff
@@ -0,0 +1,111 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import (
+	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+// ExtAuthServiceType specifies the types of External Authorization.
+// +kubebuilder:validation:Enum=GRPC;HTTP
+type ExtAuthServiceType string
+
+const (
+	// GRPC external authorization service.
+	GRPC ExtAuthServiceType = "GRPC"
+
+	// HTTP external authorization service.
+	HTTP ExtAuthServiceType = "HTTP"
+)
+
+// ExtAuth defines the configuration for External Authorization.
+type ExtAuth struct {
```

Comment:
@arkodg @envoyproxy/gateway-maintainers Could you please take a look at the ext auth API? I'd like to reach agreement on the API before moving forward with this

Reply:
can you open an API only PR so that PR can be marked as ready and can be reviewed by others as well ? tia

Reply:
https://github.com/envoyproxy/gateway/pull/2435 @arkodg